client/src/controllers/MessagesController.ts

```diff
@@ -35,56 +23,99 @@ export class MessagesController extends Controller<HTMLElement> {
   declare readonly templateTargets: HTMLTemplateElement[];
 
   declare showDelayValue: number;
+  declare element: HTMLElement;
+  declare clearDelayValue: number;
 
-  add(
-    event?: CustomEvent<{
-      /** Flag for clearing or stacking messages */
-      clear?: boolean;
-      /** Content for the message, HTML not supported. */
-      text?: string;
-      /** Message status level, based on Django's message types. */
-      type?: 'success' | 'error' | 'warning' | string;
-    }>,
-  ) {
-    const { clear = false, text = '', type } = event?.detail || {};
-
-    if (this.hasAddedClass) {
-      this.element.classList.add(this.addedClass);
-    }
+  static afterLoad() {
+    window.handleUnsavedEvent = (event) => {
+      const { formDirty, commentsDirty } = event.detail;
+      const anyDirty = formDirty || commentsDirty;
+      const allDirty = formDirty && commentsDirty;
+      const type = allDirty ? "all" : commentsDirty ? "comments" : "edits";
+      const detail = { type };
+      const eventName = anyDirty ? "w-unsaved:show" : "w-unsaved:hide";
+      document.dispatchEvent(new CustomEvent(eventName, { detail }));
+    };
+  }
```

Comment:
```suggestion
    static afterLoad() {
    const updateFooterSaveWarning = (formDirty, commentsDirty) => {
      const anyDirty = formDirty || commentsDirty;
      const type = [
        formDirty && commentsDirty && 'all',
        commentsDirty && !formDirty && 'comments',
        formDirty && !commentsDirty && 'edits',
      ].find(Boolean);

      const eventName = anyDirty ? 'w-unsaved:show' : 'w-unsaved:hide';
      document.dispatchEvent(new CustomEvent(eventName, { detail: { type } }));
    };

    /**
     * Ensure we have an ability to support the existing global function until
     * that functionality is migrated to Stimulus also.
     */
    window.updateFooterSaveWarning = updateFooterSaveWarning;
  }
```

This was good, but the `handleUnsavedEvent` function name did not make sense, instead we want to replace the existing `handleUnsavedEvent` with this custom function.